src/puppetlabs/puppetdb/middleware.clj

```diff
@@ -31,23 +31,21 @@
 (defn wrap-with-authorization
   "Ring middleware that will only pass through a request if the
   supplied authorization function allows it. Otherwise an HTTP 403 is
-  returned to the client.
-
-  `authorizer` is expected to take a single argument, the current
-  request. The request is allowed only if the return value of
-  `authorizor` is :authorized; otherwise, its value is taken to be a
-  message describing the reason that access is not allowed."
-  [app authorizer]
-  (let [authorizer (if-not (nil? authorizer)
-                     authorizer
-                     (constantly :authorized))]
-    (fn [req]
-      (let [auth-result (authorizer req)]
-        (if (= :authorized auth-result)
-          (app req)
-          (-> (str "Permission denied: " auth-result)
-              (rr/response)
-              (rr/status http/status-forbidden)))))))
+  returned to the client.  If get-authorizer is nil or false, all
+  requests will be accepted.  Otherwise it must accept no arguments
+  and return an authorize function that accepts a request.  The
+  request will be allowed only if authorize returns :authorized.
+  Otherwise, the return value should be a message describing the
+  reason that access was denied."
+  [app get-authorizer]
+  (fn [req]
+    (let [authorize (and get-authorizer (get-authorizer))
+          auth-result (if authorize (authorize req) :authorized)]
```

Comment:
@ajroetker It would annoy me to pass around whitelisted certnames too, what about [this](https://github.com/senior/puppetdb/commit/00ebdc61cd839a57aff420134b3822fce5492af5] instead?